api/src/main/java/javax/config/spi/ConfigSource.java

```diff
@@ -131,4 +134,14 @@ default int getOrdinal() {
      */
     String getName();
 
+    /**
+     * This callback should get invoked if an attribute change got detected.
+     *
+     * @param reportAttributeChange will be set by the {@link ConfigSourceProvider} after this
+     *                              {@code ConfigSource} got created and before any configured values
+     *                              get served.
+     */
+    default void setOnAttributeChange(Consumer<Set<String>> reportAttributeChange) {
```

Comment:
I'd rename it to setOnPropertyChanges() as it seems to be more consistent with the terminology we uses (and the fact that the method returns many keys)